common/djangoapps/entitlements/utils.py

```diff
@@ -0,0 +1,40 @@
+from course_modes.models import CourseMode
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview
+
+
+def is_course_run_entitlement_fullfillable(course_run_id, compare_date, entitlement):
+    """
+    Checks that the current run meets the following criteria for an entitlement
+
+    1) Is currently running or start in the future
+    2) A User can enroll in
+    3) A User can upgrade to the entitlement mode
+
+    Arguments:
+        course_run_id (String): The id of the Course run that is being checked.
+        compare_date: The date and time that we are comparing against.  Generally the current date and time.
```

Comment:
Is compare_date always Now except for in testing? We may want to just calculate now in the function, and update tests that call this function to mock out the date for Now.

Reply:
This creates the need to calculate and retrieve now on every call.  The other use of this method is to call it in a loop of runs.  It seemed inefficient to get now for every run and call of this method.

Reply:
I have another option that Harry recommended that I will add.  I will default the param to be now that way it can be set when needed.